trin-execution/src/execution.rs

```diff
@@ -144,13 +149,34 @@ impl State {
         )?;
 
         // execute transactions
+        let mut cumulative_gas_used = U256::ZERO;
+
         let transactions = block_tuple
             .body
             .body
             .transactions()
             .map_err(|err| Error::msg(format!("Error getting transactions: {err:?}")))?;
-        for transaction in transactions {
-            self.execute_transaction(transaction, &env)?;
+        for (index, transaction) in transactions.into_iter().enumerate() {
```

Comment:
If I'm not mistaken, we are updating and committing all data after every transaction, but not the main account trie.
This is done later at line 212 when we call `self.get_root_with_trie_diff()`. Or am I missing something?

Now, It might be easier to commit all trie changes after every transaction as well, but I don't think we should do that (I expect that it will slow things down a lot).
Not sure how easy it is, but I believe that we should execute and verify entire block (keep all changes in memory) and then commit all changes (also implying that `ExecutionPosition` shouldn't track `transaction_index`, or at least not store it).

Reply:
> keep all changes in memory 

this would be implementing a block execution cache and is out of scope of this PR.

this is more of a performance improvement then anything else

Reply:
I do think this is or something like it is required eventually, but were are not there yet

Reply:
https://github.com/ethereum/trin/issues/1333

Me and milos talked on a call what was outlined is basically required in the future, after the call we hit performance bottle necks so I will probably look into this after a metrics PR